prettyprinter/src/Prettyprinter/Internal.hs

```diff
@@ -452,6 +455,15 @@ instance Pretty a => Pretty (Maybe a) where
 -- Manually use @'hardline'@ if you /definitely/ want newlines.
 instance Pretty Text where pretty = vsep . map unsafeTextWithoutNewlines . T.splitOn "\n"
 
+-- | Convert text to Doc. Must not contain newlines. 
+-- Useful when dealing with wide characters and emojis
+unsafeTextWithLength :: Text -> Int -> Doc ann
+unsafeTextWithLength txt l = Text l txt
+
+-- | identical to the strict version
+unsafeLazyTextWithLength :: Lazy.Text -> Int -> Doc ann
+unsafeLazyTextWithLength  txt l = Text l (Lazy.toStrict txt)
```

Comment:
What's the use case for this function? I don't imagine that wide characters and emojis will typically exist in the form of a lazy Text values?!

If there's no obvious use case, I'd prefer not to add this function.